Framework/Geometry/src/Objects/MeshObject.cpp

```diff
@@ -58,8 +58,8 @@ const Kernel::Material MeshObject::material() const { return m_material; }
 */
 bool MeshObject::hasValidShape() const {
   // May enclose volume if there are at
-  // at least 4 triangles and 4 vertices
-  return (numberOfTriangles() >= 4 && numberOfVertices() >= 4);
+  // at least 2 triangles and 4 vertices
+  return (numberOfTriangles() >= 2 && numberOfVertices() >= 4);
```

Comment:
4 triangles & 4 vertices is the simplest MeshObject that can enclose a volume. Can you please explain why the number of triangles has been reduced to 2 while the number of vertices remains 4?

Reply:
If this fixes a bug, how does one produce the bug?

Reply:
@KarlPalmen I believe this would constitute a more robust check for a closed shape https://en.wikipedia.org/wiki/Euler_characteristic. We load a completely planar mesh with >> 4 faces and vertices and the check passes. This would obviously cause problems since it does not enclose a volume. I think the current check is insufficient.

Reply:
What I see above is a less robust check. 

It could be followed by further checks such as the Euler characteristic check, although even that would not work for a surface enclosing multiple volumes.
I decided not the put a more robust check in, because it would require considerable effort. I did consider the Euler characteristic check. It would require determining and counting the edges. Also on the way one could check each edge has the 2 vertices in opposite order in the two triangles that form the edge. Another less robust possibility is to check the the volume is positive.

You are welcome to add such checks, but there impact on performance needs testing.